front/app/containers/Admin/projects/project/ideas/AnalysisBanner.tsx

```diff
@@ -13,12 +13,26 @@ import { useIntl } from 'utils/cl-intl';
 import messages from './messages';
 import { useParams } from 'react-router-dom';
 import clHistory from 'utils/cl-router/history';
+import useAnalyses from 'api/analyses/useAnalyses';
+import useAddAnalysis from 'api/analyses/useAddAnalysis';
 
 const AnalysisBanner = () => {
   const { projectId } = useParams() as { projectId: string };
+  const { data: analyses } = useAnalyses({ projectId });
+  const { mutate: createAnalysis } = useAddAnalysis();
+
   const { formatMessage } = useIntl();
+
   const analysisEnabled = useFeatureFlag({ name: 'analysis' });
+
+  useEffect(() => {
+    if (analysisEnabled && analyses && analyses.data.length === 0) {
+      createAnalysis({ projectId });
+    }
+  }, [analyses, createAnalysis, projectId, analysisEnabled]);
```

Comment:
If I get it right, we're creating the analysis even before the user hits the button, so at the moment the banner is first displayed, correct?

We haven't properly discussed how to deal with updates to the data, but this would make one (currently) problematic situation more common: A custom field is added to the form after the analysis has been created. Or in general, any kind of functionality that properly deals with data changes (new inputs, form changes, ...) will need to work well from the start, as it will very often initialize empty.

I know we will need to take care of these things anyway, but to not make current limitations worse, could we for now only create when the user clicks the button?

Reply:
Yes, we can 👍